### PR TITLE
IT-1673: deploy BDC

### PR DESCRIPTION
**Pull request by adrienthebo**

### CODE REVIEW

File: hieradata/site/ls.yaml [hunk 1]
```diff
@@ -1,110 +1,143 @@
...
+resolv_conf::nameservers: *nameservers
+resolv_conf::searchpath: *dnsdomains
+
+sudo::config_file_replace: false
+sudo::purge: false
+sudo::configs:
+  wheel_b:  # backup wheel group in case of LDAP failure.
+    content: "%wheel_b ALL=(ALL) NOPASSWD: ALL"
+
+accounts::group_list:
```
Comment (adrienthebo) [inline hieradata/site/ls.yaml]: This needs to be dropped.

File: hieradata/site/ls.yaml [hunk 2]
```diff
@@ -1,110 +1,143 @@
...
+  - "1.cl.pool.ntp.org"
+  - "1.south-america.pool.ntp.org"
+
+chrony::servers: *ntpservers
+resolv_conf::nameservers: *nameservers
+resolv_conf::searchpath: *dnsdomains
+
+sudo::config_file_replace: false
+sudo::purge: false
+sudo::configs:
```
Comment (adrienthebo) [inline hieradata/site/ls.yaml]: This needs to be dropped.

File: hieradata/cluster/corels.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+---
+firewall::ensure: "stopped"
```
Comment (adrienthebo) [inline hieradata/cluster/corels.yaml]: This should be pushed to the entire site.

Comment (jhoblitt) [reply hieradata/cluster/corels.yaml]: Agreed. I'll open a PR to move come values up from the cluster level.

File: hieradata/site/ls/role/ipamaster.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+---
+easy_ipa::ipa_master_fqdn: "ipa1.cp.lsst.org"
```
Comment (jhoblitt) [inline hieradata/site/ls/role/ipamaster.yaml]: Should this be `ipa1.ls.lsst.org`?

File: hieradata/site/ls/role/ipareplica.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+easy_ipa::configure_replica_ca: true
+easy_ipa::ipa_master_fqdn: "ipa1.cp.lsst.org"
```
Comment (jhoblitt) [inline hieradata/site/ls/role/ipareplica.yaml]: Should this be ipa1.ls.lsst.org?

File: hieradata/site/ls/role/ipamaster.yaml (hunk 4 above)
Comment (adrienthebo) [reply hieradata/site/ls/role/ipamaster.yaml]: No, this is the IPA master that the replica attaches to.

File: hieradata/site/ls/role/ipareplica.yaml (hunk 5 above)
Comment (adrienthebo) [reply hieradata/site/ls/role/ipareplica.yaml]: As above, this is the IPA master that the replica attaches to.

File: hieradata/cluster/corels.yaml (hunk 3 above)
Comment (jhoblitt) [reply hieradata/cluster/corels.yaml]: https://github.com/lsst-it/lsst-itconf/pull/41

File: hieradata/site/ls.yaml [hunk 6]
```diff
@@ -1,110 +1,143 @@
...
-rsyslog::client::global_config:
+lookup_options:
+  accounts::user_list:
+    merge:
+      strategy: "deep"
+
+easy_ipa::ipa_master_fqdn: "ipa1.ls.lsst.org"
+
+# Rsyslog configuration is based on the default rsyslog.conf shipping with CentOS 7.7
+rsyslog::config::global_config:
```
Comment (jhoblitt) [inline hieradata/site/ls.yaml]: I don't want to make it difficult to get this PR merged but is there any reason we can't set the rsyslog configuration at the org level?  There is nothing working in Tucson now.

File: hieradata/site/ls/role/ipareplica.yaml (hunk 5 above)
Comment (jhoblitt) [reply hieradata/site/ls/role/ipareplica.yaml]: Doh, yes, of course.  However, I believe `ipa1.tuc.lsst.org` is currently the master?

File: hieradata/site/ls.yaml (hunk 6 above)
Comment (adrienthebo) [reply hieradata/site/ls.yaml]: Implemented.

**Comment by adrienthebo** [comment]
Applied changes:

* firewall disabled for `ls` site
* rsyslog global configuration pushed to `org/lsst.yaml`
* site specific sudo and accounts dropped in favor of `org/lsst` definitions

This is ready for another review, @jhoblitt.

File: hieradata/site/ls/role/ipareplica.yaml (hunk 5 above)
Comment (adrienthebo) [reply hieradata/site/ls/role/ipareplica.yaml]: Once a domain replica has been added, it has the same rights and capabilities as any other master. `man ipa-replica-install` says the following:

```
       ipa-replica-install - Create an IPA replica

SYNOPSIS
   DOMAIN LEVEL 0
       ipa-replica-install [OPTION]... [replica_file]

   DOMAIN LEVEL 1
       ipa-replica-install [OPTION]...

DESCRIPTION
       Configures  a new IPA server that is a replica of the server. Once it has been created it is an
       exact copy of the original IPA server and is an equal master. Changes made to  any  master  are
       automatically replicated to other masters.
```

Right now the `139.229.135.0/24` subnet can't directly reach ipa1.tuc.lsst.cloud so I've attached ipa1.ls.lsst.org to ipa1.cp.lsst.org. I've promoted ipa1.cp.lsst.org to a CA master as well and have replicated the CA to ipa1.ls.lsst.org from there.

Screenshot from 2020-02-20 15-37-05

Comment (jhoblitt) [reply hieradata/site/ls/role/ipareplica.yaml]: I was not aware that the CA has been replicated. I've opened https://jira.lsstcorp.org/browse/IT-1796 to make sure this becomes part of the puppet configuration.

Comment (adrienthebo) [reply hieradata/site/ls/role/ipareplica.yaml]: This is being handled by Puppet on new nodes:

https://github.com/lsst-it/lsst-itconf/pull/40/files#diff-032e052f37da931d7d6e365ac337d9ddR2

**Pull request merged**